### ISSUE TITLE
Optional subcommands fail with a `KeyError`

### ISSUE THREAD

**Issue by carmocca** [omni-us/jsonargparse#68]
```python
from jsonargparse import ArgumentParser

parser = ArgumentParser()
subcommands = parser.add_subcommands(required=False)
subparser = ArgumentParser()
subcommands.add_subcommand('foo', subparser)
parser.parse_args()
```

When called with no subcommand, prints:

```python
Error: None
```

After some debugging, I found two issues:

(1):

https://github.com/omni-us/jsonargparse/blob/e7ab0f3c2e687563a5f5e8eca8087f61b39977dd/jsonargparse/actions.py#L767

should be replaced with 

```python
        subparser = action._name_parser_map.get(subcommand)
        if subparser is None:
            return
```

(2):

KeyErrors should not be catched here:

https://github.com/omni-us/jsonargparse/blob/e7ab0f3c2e687563a5f5e8eca8087f61b39977dd/jsonargparse/core.py#L396-L397

which hides the original exception when it seems to be an actual bug:

```python
Traceback (most recent call last):
  File "/Users/carlosmocholi/Library/Application Support/JetBrains/PyCharmCE2020.3/scratches/scratch_12.py", line 8, in <module>
    parser.parse_args()
  File "/Users/carlosmocholi/.pyenv/versions/3.9.1/envs/lightning-3.9/lib/python3.9/site-packages/jsonargparse/core.py", line 385, in parse_args
    parsed_cfg = self._parse_common(
  File "/Users/carlosmocholi/.pyenv/versions/3.9.1/envs/lightning-3.9/lib/python3.9/site-packages/jsonargparse/core.py", line 308, in _parse_common
    _ActionSubCommands.handle_subcommands(self, cfg, env=env, defaults=defaults, fail_no_subcommand=fail_no_subcommand)
  File "/Users/carlosmocholi/.pyenv/versions/3.9.1/envs/lightning-3.9/lib/python3.9/site-packages/jsonargparse/actions.py", line 767, in handle_subcommands
    subparser = action._name_parser_map[subcommand]
KeyError: None
```

**Comment by mauvilsa** [comment]
The fix is now in master.
